tests/parser/parser.cpp

```diff
@@ -595,7 +595,7 @@ TEST(Parser, modelWithInvalidUnits)
         "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
         "<model xmlns=\"http://www.cellml.org/cellml/2.0#\" name=\"model_name\">\n"
         "  <units name=\"fahrenheitish\">\n"
-        "    <unit exponent=\"inf\" units=\"kelvin\"/>\n"
+        "    <unit exponent=\"nan\" units=\"kelvin\"/>\n"
```

Comment:
Why is the `@exponent` changing in this case? the source value is `35.0E+310` which is presumably `inf` not `nan`?

Reply:
We use `std::stod()` to convert a `std::string` to a `double`, and either the conversion is successful or it is not.

If it is not, it is either because:
 - an invalid argument was passed; or because
 - the passed value is out of range.

This is likely to be the latter here, but then we can't tell whether it should be `+inf` or `-inf` unless we were to check the sign of the passed value...?

So, maybe we could do the following: try to convert using `std::stod()` and if we get no exception then it's all good otherwise if we get:
 - an `invalid_argument` exception then we return `nan`; or if we get
 - an `out_of_range` exception then we check whether the passed value starts with `-`. If it does then we return `-inf` otherwise we return `inf`.

Would that be better?